crates/core/plugin_sm/src/plugin.rs

```diff
@@ -462,7 +462,7 @@ impl Plugin for ExternalPluginCommand {
                     self.name.clone(),
                     match last_line {
                         0 => output.stdout.as_slice(),
-                        _ => &output.stdout[..=last_line],
+                        _ => &output.stdout[..=last_line - 1],
```

Comment:
The limitation of `max_packages -1` must be done here.

If we applied it there `.nth(usize::try_from(self.max_packages - 1).unwrap_or(usize::MAX))` there would be one case that doesn't work: If user set `max_packages = 1` then `last_line` would be 0 and we would return whole list instead of one package.

Reply:
Actually this didn't work as expected because the variable name is misleading as it should be `last_char` and not `last_line` as the it is taking a slice of the output which is just a vector of characters (not a vector of lines).

I reworked the example to make it a bit more clear, and there were some other minor refactoring to make the code a bit more readable.